test/runtime_tests.cc

```diff
@@ -1228,6 +1229,55 @@ namespace detail {
 		dry_run_with_nodes(num_nodes);
 	}
 
+	TEST_CASE_METHOD(test_utils::runtime_fixture, "Dry run proceeds on fences", "[dryrun]") {
+		env::scoped_test_environment ste(std::unordered_map<std::string, std::string>{{dryrun_envvar_name, "1"}});
+
+		distr_queue q;
+
+		auto& rt = runtime::get_instance();
+		auto& tm = rt.get_task_manager();
+
+		REQUIRE(rt.is_dry_run());
+
+		buffer<bool, 0> buf{false};
+		q.submit([&](handler& cgh) {
+			accessor acc{buf, cgh, all{}, write_only_host_task};
+			cgh.host_task(on_master_node, [=] { acc = true; });
+		});
+
+		auto ret = experimental::fence(q, buf);
```

Comment:
Maybe add a TODO here that we should check for the warning being printed once we've sorted out the issues with `log_capture`.